edgeworkers/Acrobat_DC_web_prod/main.js

```diff
@@ -162,22 +172,28 @@ export async function responseProvider(request) {
   };
 
   try {
-    const [
-      [responseStream, responseHeaders, dcCoreVersion],
-    ] = await Promise.all([
-      fetchFrictionlessPageAndInlineSnippet(),
-      fetchAndInlineScripts(),
+    const [responseStream, responseHeaders, dcCoreVersion, mobileWidget] = await fetchFrictionlessPageAndInlineSnippet();
+
+    await Promise.all([
+      fetchAndInlineScripts(mobileWidget),
```

Comment:
looks like this will hurt our performance. There are fetches in there that would otherwise be done in parallel with the first request

Reply:
We need to know if `mobile-widget` is true or false to decide what scripts to be inline and preloads. Therefore, they have to be separated.

Reply:
scripts.js is loaded regardless.

loading dc-conversion-widget.js conditionally, blocked by the first Promise, will hurt Frictionless performance on production where it is used. It might be better to load it, in parallel, and not use it where you don't need it.

Reply:
perhaps we could break fetchAndInlineScripts into two separate methods - fetchScript and inlineScripts. Then we could still fetch the frictionlesspage and fetchscripts in parallel, and then call inlineScripts(mobileWidget)